tests/common/helpers/snmp_helpers.py

```diff
@@ -10,20 +10,34 @@
 
 DEF_WAIT_TIMEOUT = 300
 DEF_CHECK_INTERVAL = 10
+SNMP_SUBAGENT_WAIT_TIMEOUT = 120
+SNMP_SUBAGENT_CHECK_INTERVAL = 5
 
 global_snmp_facts = {}
 
 
+def is_snmp_subagent_running(duthost):
+    cmd = "docker exec snmp supervisorctl status snmp-subagent"
+    output = duthost.shell(cmd)
+    if "RUNNING" in output["stdout"]:
+        return True
+    return False
+
+
 def _get_snmp_facts(localhost, host, version, community, is_dell, include_swap, module_ignore_errors):
     snmp_facts = localhost.snmp_facts(host=host, version=version, community=community, is_dell=is_dell,
                                       module_ignore_errors=module_ignore_errors, include_swap=include_swap)
     return snmp_facts
 
 
-def _update_snmp_facts(localhost, host, version, community, is_dell, include_swap):
+def _update_snmp_facts(localhost, host, version, community, is_dell, include_swap, duthost):
     global global_snmp_facts
 
     try:
+        pytest_assert(
+            wait_until(SNMP_SUBAGENT_WAIT_TIMEOUT, SNMP_SUBAGENT_CHECK_INTERVAL, 0,
```

Comment:
If Sub-Agent isn't a prerequisites of getting snmp facts, then it explains.
But let's not call wait_until(b) in another wait_until(a), that makes the first result of wait_until(a) to hanging for a long time.

Suggest:
```
#Current
def _update_snmp_facts():
    wait_until(is_snmp_subagent_running)
    global_snmp_facts = _get_snmp_facts(localhost, host, version, community, is_dell, include_swap,
                                            module_ignore_errors=False)
    return True when no exception
```
to
```
def _update_snmp_facts():
    snmp_subagent_running =is_snmp_subagent_running()
    global_snmp_facts = _get_snmp_facts(localhost, host, version, community, is_dell, include_swap,
                                            module_ignore_errors=False)
    return snmp_subagent_running and (True when no exception)
```

Reply:
added the change suggested.